webui/react/src/pages/Admin/UserManagement.settings.ts

```diff
@@ -30,68 +29,51 @@ export const DEFAULT_COLUMN_WIDTHS: Record<UserColumnName, number> = {
   modifiedAt: 80,
 };
 
-export interface UserManagementSettings extends InteractiveTableSettings {
-  name?: string;
-  sortDesc: boolean;
-  sortKey: V1GetUsersRequestSortBy;
-}
+export const UserStatus = {
+  ACTIVE: 'active',
+  INACTIVE: 'inactive',
+} as const;
+export type UserStatus = ValueOf<typeof UserStatus>;
 
-const config: SettingsConfig<UserManagementSettings> = {
-  settings: {
-    columns: {
-      defaultValue: DEFAULT_COLUMNS,
-      skipUrlEncoding: true,
-      storageKey: 'columns',
-      type: array(string),
-    },
-    columnWidths: {
-      defaultValue: DEFAULT_COLUMNS.map((col: UserColumnName) => DEFAULT_COLUMN_WIDTHS[col]),
-      skipUrlEncoding: true,
-      storageKey: 'columnWidths',
-      type: array(number),
-    },
-    name: {
-      defaultValue: undefined,
-      storageKey: 'name',
-      type: union([string, undefinedType]),
-    },
-    row: {
-      defaultValue: undefined,
-      skipUrlEncoding: true,
-      storageKey: 'row',
-      type: union([array(number), array(string), undefinedType]),
-    },
-    sortDesc: {
-      defaultValue: true,
-      storageKey: 'sortDesc',
-      type: boolean,
-    },
-    sortKey: {
-      defaultValue: V1GetUsersRequestSortBy.MODIFIEDTIME,
-      storageKey: 'sortKey',
-      type: union([
-        literal(V1GetUsersRequestSortBy.ACTIVE),
-        literal(V1GetUsersRequestSortBy.ADMIN),
-        literal(V1GetUsersRequestSortBy.DISPLAYNAME),
-        literal(V1GetUsersRequestSortBy.MODIFIEDTIME),
-        literal(V1GetUsersRequestSortBy.UNSPECIFIED),
-        literal(V1GetUsersRequestSortBy.USERNAME),
-        literal(V1GetUsersRequestSortBy.NAME),
-        literal(V1GetUsersRequestSortBy.LASTLOGINTIME),
-      ]),
-    },
-    tableLimit: {
-      defaultValue: MINIMUM_PAGE_SIZE,
-      storageKey: 'tableLimit',
-      type: number,
-    },
-    tableOffset: {
-      defaultValue: 0,
-      storageKey: 'tableOffset',
-      type: number,
-    },
-  },
-  storagePath: 'user-management',
-};
+export const UserRole = {
+  ADMIN: 'admin',
+  MEMBER: 'member',
+} as const;
+export type UserRole = ValueOf<typeof UserRole>;
 
-export default config;
+export const UserManagementSettings = t.intersection([
+  t.type({
+    sortDesc: t.boolean,
+    tableLimit: t.number,
+    tableOffset: t.number,
```

Comment:
question: do we wanna store the `tableOffset`? I feel strange when tableOffset is stored even after moving back from other pages

Reply:
let's keep it for now and remove it after clearing with product.